src/lib.rs

```diff
@@ -493,10 +493,16 @@ impl Pixels {
 
     /// Get a mutable byte slice for the pixel buffer. The buffer is _not_ cleared for you; it will
     /// retain the previous frame's contents until you clear it yourself.
-    pub fn get_frame(&mut self) -> &mut [u8] {
+    pub fn get_frame_mut(&mut self) -> &mut [u8] {
         &mut self.pixels
     }
 
+    /// Get an immutable byte slice for the pixel buffer. The buffer is _not_ cleared for you; it
+    /// will retain the previous frame's contents until you clear it yourself.
```

Comment:
Let's remove the note about clearing the buffer from this one. The slice is immutable so there is no way to clear the buffer through it.

I would also go as far to say that the doc comment should specify more of its purpose. It seems to me the only point of this is to treat the pixel buffer as the source of truth for what will be rendered. While I never really envisioned it being used that way, I can see the merits. Sort of. I personally don't have any reason to want to read from the buffer, but maybe someone out there does. Blending in software or something?

Reply:
Blending precisely – I'm using this for compositing. I clear the buffer at the start of each frame, so at any point during rendering the pixel buffer represents what has been rendered below the current thing that is being rendered.

I'll update the comment to reflect this.